### ISSUE TITLE
Coding style

### ISSUE THREAD

**Issue by xiangzhu** [stephenslab/rssr#9]
I think it is time to think about some style guide for `R` codes in this package at least.

- https://google.github.io/styleguide/Rguide.xml
- http://adv-r.had.co.nz/Style.html
- https://www.bioconductor.org/developers/how-to/coding-style/

**Comment by pcarbo** [comment]
I agree that it is important to set some coding standards. And this is a good starting point. But I think that there are other important aspects to coding style that aren't covered by these guides. In other words, you could follow these style guides very carefully and still have code that is difficult to follow. And coding style standards can become never-ending, e.g.,

https://github.com/twitter/commons/blob/master/src/java/com/twitter/common/styleguide.md

A useful approach is to do "peer review" when coding. For example, if you implement a new function, have Nick look at it and makes sure he understands what the code is doing; any parts that are unclear should be improved with better variable names, more comments, _etc._

**Comment by xiangzhu** [comment]
@pcarbo  Thanks for the suggestion!

Nick and I will soon start this "peer review" process -- Nick is the "author" and I am the "reviewer" at this moment, since Nick writes all the `R` codes.

**Comment by xiangzhu** [comment]
Hadley Wickham's coding style:

http://style.tidyverse.org/index.html

There is also a package associated with this style:

https://github.com/jimhester/lintr
